app/controllers/api/v1/message_create_controller.rb

```diff
@@ -0,0 +1,33 @@
+module Api
+  module V1
+    class MessageCreateController < Api::V1::ApplicationController
+      before_action :users, only: [:create]
+      before_action :message_sent, only: [:create]
+
+      def create
+        @messages = Message.where(recipient_id: @recipient.id, sender_id: @current_user.id)
+                           .or(Message.where(sender_id: @recipient.id, recipient_id: @current_user.id))
+        @old_time = I18n.l(Message.last.created_at, format: :short)
+        ActionCable.server.broadcast('MyChannel',
+                                     { body: params[:body], current_user_id: @current_user.id, created_at: @old_time,
+                                       messages: @messages })
+      end
+
+      def users
+        @recipient = User.find params[:user_id]
+        @current_user = User.find params[:current_user_id]
+      end
+
+      def message_sent
+        @message = @current_user.sent.create(recipient: @recipient, body: params[:body])
```

Comment:
add a custom validator to message body - minimum words number 2

before you save message do the following:

1. replace all the 'bad' words with *
2. escape all the javascript